phpBB/styles/prosilver/template/mcp_jumpbox.html

```diff
@@ -0,0 +1,18 @@
+<div class="action-bar actions-jump">
+	<div class="jumpbox dropdown-container dropdown-container-right<!-- IF not S_IN_MCP --> dropdown-up<!-- ENDIF --> dropdown-{S_CONTENT_FLOW_BEGIN} dropdown-button-control" id="jumpbox">
+			<span title="<!-- IF S_IN_MCP and S_MERGE_SELECT -->{L_SELECT_TOPICS_FROM}<!-- ELSEIF S_IN_MCP -->{L_MODERATE_FORUM}<!-- ELSE -->{L_JUMP_TO}<!-- ENDIF -->" class="button button-secondary dropdown-trigger dropdown-select">
+				<span><!-- IF S_IN_MCP and S_MERGE_SELECT -->{L_SELECT_TOPICS_FROM}<!-- ELSEIF S_IN_MCP -->{L_MODERATE_FORUM}<!-- ELSE -->{L_JUMP_TO}<!-- ENDIF --></span>
+				<span class="caret"><i class="icon fa-sort-down fa-fw" aria-hidden="true"></i></span>
+			</span>
+		<div class="dropdown">
+			<div class="pointer"><div class="pointer-inner"></div></div>
+			<ul class="dropdown-contents">
+				<!-- BEGIN jumpbox_forums -->
```

Comment:
why are we not using the twig function for this? Or have you just not gotten to it yet?

Reply:
I think is better to use another thing for this, because we had two diferent jumpbox in one single function.
They are similar, they can use the same forum_provider or whatever (when its done), but don't use both in the same function.
Anyway, i think this pr was just a concept.

Reply:
sure im just not sure why they need to be different. I have not gotten there yet with the new theme . Granted im not using the db scheme from prosilver for my testsfor chameleon, but the way I am handling it is far simpler https://github.com/phpbb/chameleon/blob/master/all/views/includes/jumpto.twig

Reply:
That looks much better than in prosilver.
But the purpose of this was to propose a way to remove the make_jumpbox function from all controllers.

Reply:
ok got it, but we still need to maintain consistency across the theme in the way that we are creating the jumpbox. This is probably something that needs flagged as a bug then for the fron-end to be handle first. As again not sure why we are doing this differently in different places possibly an oversight when jumpbox.html was created

Reply:
Just close this if you are going to do it in another way.

Reply:
wait Im still confused sorry I do not have access to a code editor so I m stuck with only github at my disposal. What exactly id the differrence between jumpbox.html and mcp_jumpbox.html besides the wrapping div you left in mcp_jumpbox.html

Reply:
In jumpbox.html i removed all the code related to the mcp.
The diference is that one is the normal jumpbox to jump between forums, and the other is to jump between moderate forums in mcp (one displays the forums the user have permission to see, and the other display the forums the user have permission to moderate). I think both jumpbox should share code/dependencies, but they shouldn't be the same, because the code becomes complex (but i can be wrong)
Also there is another jumpbox that isn't a jumpbox, that is the one used in mcp.

Normal jumpbox
https://i.imgur.com/KIRdigj.png

Mcp jumpbox:
https://i.imgur.com/U9oZNPK.png

Select that isn't a jumpbox, but do the same:
https://i.imgur.com/2oICfKB.png

Reply:
ok I gotcha yeah its not really overly complex looks like the only differencs is the title and the css classes which can both be passed in as paramaters to the function.

`JumpBox(css_classes, title = lang('JUMP_TO'), )`

`lang('DIRECTION') == 'ltr' ? 'left' : 'right'` is not needed just use {{ S_CONTENT_FLOW_BEGIN }}

something like

```
<div class="jumpbox dropdown-container dropdown-container-righ dropdown-{{ S_CONTENT_FLOW_BEGIN }}} dropdown-button-control{% css_classes ? css_classes : '' %}" id="jumpbox">
			<span title="{{ title }}" class="button button-secondary dropdown-trigger dropdown-select">
				<span{{ title }}</span>
				<span class="caret"><i class="icon fa-sort-down fa-fw" aria-hidden="true"></i></span>
			</span>
		<div class="dropdown">
			<div class="pointer"><div class="pointer-inner"></div></div>
			<ul class="dropdown-contents">
				{% for forum in jumpbox_forums %}
				{% if forum.FORUM_ID != -1 %}
				<li>
					<a href="{{ forum.LINK }}" class="{% if forum.level %}jumpbox-sub-link{% elseif forum.S_IS_CAT %}jumpbox-cat-link{% else %}jumpbox-forum-link{% endif %}">
						{% for level in forum.level %}<span class="spacer"></span>{% endfor %}
						<span>
							{% if forum.level %}{% if lang('DIRECTION') == 'rtl' %}&#8626;{% else %}&#8627;{% endif %} &nbsp;{% endif %}
							{{ forum.FORUM_NAME }}
						</span>
				</li>
				{% endif %}
				{% endfor %}
			</ul>
		</div>
	</div>
```

Reply:
Yes, the template probably can be the same, i was saying the php function.
So finally you like this or is going to be done in other way?

Reply:
as long as you can handle it with the same template for all then yes

Reply:
Ah ok, the idea for now is wait for the index refactor and use its forum_provider (or whatever) here.
I will update the name tomorrow

Reply:
ok no worries sounds good